lint/src/test/java/permissions/dispatcher/CallNeedsPermissionDetectorTest.java

```diff
@@ -0,0 +1,86 @@
+package permissions.dispatcher;
+
+import com.google.common.collect.ImmutableList;
+
+import com.android.tools.lint.checks.infrastructure.LintDetectorTest;
+import com.android.tools.lint.detector.api.Detector;
+import com.android.tools.lint.detector.api.Issue;
+
+import org.junit.Test;
+
+import java.util.Collections;
+import java.util.List;
+
+public class CallNeedsPermissionDetectorTest extends LintDetectorTest {
+
+    private static final String NO_WARNINGS = "No warnings.";
+
+
+    @Override
+    protected Detector getDetector() {
+        return new CallNeedsPermissionDetector();
+    }
+
+    @Override
+    protected List<Issue> getIssues() {
+        return ImmutableList.of(CallNeedsPermissionDetector.ISSUE);
+    }
+
+    @Test
+    public void testCallNeedsPermissionMethod() throws Exception {
+        CallNeedsPermissionDetector.METHODS = Collections.singletonList("fooBar");
+
+        StringBuilder foo = new StringBuilder("package com.example;\n");
+        foo.append("public class Foo {\n");
+        foo.append("public void someMethod() {");
+        foo.append("Baz baz = new Baz();\n");
+        foo.append("baz.fooBar();  ");
+        foo.append("}\n");
+        foo.append("}");
+
+        StringBuilder baz = new StringBuilder("package com.example;\n");
+        baz.append("public class Baz {\n");
+        baz.append("public void fooBar() {\n");
+        baz.append("}\n");
+        baz.append("}");
```

Comment:
It was a bit messy concatenating them together using `+` because of the code formatter.